.github/ISSUE_TEMPLATE/01_bug_report.yml

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Bug Report
+description: Create a report to help us improve Bolt.
+title: "[Bug] "
+labels: ["bug", "needs triage"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        > Please search existing issues to avoid duplicates.
+        > For security issues, please contact us privately instead of posting here.
+
+  - type: checkboxes
+    attributes:
+      label: Component Selection
+      description: Which part of Bolt is affected?
+      options:
+        - label: Core Engine (Expression eval, Memory, Vector)
+        - label: Connectors (Hive, Parquet, etc.)
+        - label: API / Bindings (JNI, Python, etc.)
+        - label: Build / Installation
+        - label: Other
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Describe the Bug
+      description: |
+        A clear and concise description of what the bug is.
+      placeholder: |
+        When I run query X with data Y, the engine crashes / returns wrong result...
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Reproduction Steps
+      description: |
+        Steps to reproduce the behavior. 
+        **Crucial**: Please provide a Minimal Reproducible Example (SQL query, Unit Test, or Data Schema).
+      placeholder: |
+        1. Create table with schema...
+        2. Run query: select * from ...
+        3. See error...
+    validations:
+      required: true
+
+  - type: input
+    id: version
+    attributes:
+      label: Bolt Version / Commit ID
+      description: e.g. v0.1.0 or commit hash `a1b2c3d`
+      placeholder: main branch @ commit ...
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: System Configuration (Critical for C++)
+      description: |
+        Please provide details about your environment.
+      value: |
+        - **OS**: (e.g. Ubuntu 22.04, CentOS 7)
+        - **Compiler**: (e.g. GCC 11, Clang 14)
+        - **Build Type**: (Debug / Release / RelWithDebInfo)
+        - **CPU Arch**: (e.g. x86_64 AVX2, ARM64)
```

Comment:
Maybe we should instead provide some kind of script to collect the environment information in a nice copy-and-pastable format

Reply:
That's a valid point for build issues. I will add a command in Makefile to help collect build info, but for runtime bugs, it likely won't work. 

Many runtime bugs are reported by end-users or testers who are running the distributed binary on a clean environment. Asking them to install scripts just to generate an environment report is not realistic. But maybe we can add a script to the repo later as an optional helper for users who are willing to run it

Reply:
I have done this